crates/integrations/datafusion/src/physical_plan/write.rs

```diff
@@ -144,6 +144,16 @@ impl ExecutionPlan for IcebergWriteExec {
         self
     }
 
+    /// Prevents the introduction of additional `RepartitionExec` and processing input in parallel.
```

Comment:
This is additional improvement related to plan execution that prevents the introduction of additional `RepartitionExec`. The plan will preserve the same partitions num as input but won't partition input all the time (default target is num of CPU cores) as it seems unnecessary.